### DISCUSSION TITLE
Miss date_trunc

### DISCUSSION THREAD

**Discussion by Qzet** [dbeaver/dbeaver#12373] [Q&A]
Hi!
Have a problem on DBeaver v.21.0.4.202105021514 (community edition), macOS v.11.2.3.
I can't call date_trunc... it's just missing
Is this mb some mac restrictions? Or I need some libraries?
Pls, help with this

**Comment by LonwoLonwo** [comment]
Hello @Qzet 
What Database do you use? Is it missing in SQL Editor auto-completion? Can you provide some examples?

**Reply by Qzet** [reply]
For now I work in PostgreSQL (DBeaver v.21.0.4.202105021514 (community edition)) test database - DBeaver Sample Database (SQLite).
It's missing in auto-completion and when I type it in SQL Editor (with necessary arguments), return error:
_"SQL Error [1]: [SQLITE_ERROR] SQL error or missing database (no such function: date_trunk)"_
I don't know what examples I can give, it's just ruined on beginning. For example: _select *, date_trunc('month', BirthDate) from Employee e_

**Reply by LonwoLonwo** [reply]
So.. PostgreSQL or SQLite?
Your error is "SQLITE_ERROR". SQLite does not have "date_trunc" function.
You can use it only in PostgreSQL.

**Reply by Qzet** [reply]
Oh... I get it. Thanks for your clarify

**Reply by LonwoLonwo** [reply]
https://stackoverflow.com/questions/28638907/postgresql-sqlite-date-trunc-equivalent may be helps
